static/js/realm_icon.js

```diff
@@ -10,6 +10,9 @@ export function build_realm_icon_widget(upload_function) {
     };
 
     if (!page_params.is_admin) {
+        $(".image-delete-button").hide();
+        $(".image-upload-background").hide();
+        $(".image-upload-text").hide();
```

Comment:
We should use a more specific selector here like `$("#realm-icon-upload-widget .image-delete-button")`. And then have this code for all image upload widgets.

Reply:
Though I think instead of using this solution, we can simply hide the parent div of all these elements. We already add hide class to the `image_upload_button` element using `is_editable_by_current_user` element, it is just that the css for hide class is not applied correctly now because of the preference order of css. We would need to fix that. 

We would need some changes in CSS (maybe in templates too) apart from adding the CSS for hide class in order to make this work completely correct for all widgets.